akka-docs/src/test/java/jdocs/typed/tutorial_5/DeviceGroupQuery.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package jdocs.typed.tutorial_5;
+
+import akka.actor.typed.ActorRef;
+import akka.actor.typed.Behavior;
+import akka.actor.typed.javadsl.AbstractBehavior;
+import akka.actor.typed.javadsl.ActorContext;
+import akka.actor.typed.javadsl.Behaviors;
+import akka.actor.typed.javadsl.Receive;
+import akka.actor.typed.javadsl.TimerScheduler;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static jdocs.typed.tutorial_5.DeviceManagerProtocol.*;
+
+//#query-full
+//#query-outline
+public class DeviceGroupQuery extends AbstractBehavior<DeviceGroupQueryMessage> {
+
+  public static Behavior<DeviceGroupQueryMessage> behavior(
+    Map<String, ActorRef<DeviceProtocol.DeviceMessage>> deviceIdToActor,
+    long requestId,
+    ActorRef<RespondAllTemperatures> requester,
+    Duration timeout) {
+    return
+      Behaviors.setup(context ->
+        Behaviors.withTimers(timers -> new DeviceGroupQuery(
+          deviceIdToActor, requestId, requester, timeout, context, timers)));
+  }
+
+  private static enum CollectionTimeout implements DeviceGroupQueryMessage {
+    INSTANCE
+  }
+
+  static class WrappedRespondTemperature implements DeviceGroupQueryMessage {
+    final DeviceProtocol.RespondTemperature response;
+
+    WrappedRespondTemperature(DeviceProtocol.RespondTemperature response) {
+      this.response = response;
+    }
+  }
+
+  private static class DeviceTerminated implements DeviceGroupQueryMessage {
+    final String deviceId;
+
+    private DeviceTerminated(String deviceId) {
+      this.deviceId = deviceId;
+    }
+  }
+
+  private final long requestId;
+  private final ActorRef<RespondAllTemperatures> requester;
+  //#query-outline
+  //#query-state
+  private Map<String, TemperatureReading> repliesSoFar = new HashMap<>();
+  private Set<String> stillWaiting = new HashSet<>();
```

Comment:
final and unitialised here as done in the constructor?

Reply:
fixed